### ISSUE TITLE
RemoveEmptyJavaDocParameters causes ClassCastExceptions

### ISSUE THREAD

**Issue by AlexanderSkrock** [openrewrite/rewrite#3078]
Hi,

when I run rewrite on my current project, a couple `ClassCastException`s  occur. As side note, that project is compiled with Java 17, so I also had to run rewrite trough Maven using that JDK version. Otherwise I had a missmatch between the executing java version and the version of my compiled classes.
We just migrated to Java 17 in the recent months, before I was able to run rewrite without any problems. I expected rewrite to be compatible with Java 17 as there are also java 17 specific recipes. Or is it a problem in combination with the `rewrite-maven-plugin` which is compiled with Java 8? If so, I would raise an issue over there.

The following exceptions occured:

-  `class org.openrewrite.json.tree.Json$Document cannot be cast to class org.openrewrite.java.tree.J (org.openrewrite.json.tree.Json$Document and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`
- `class org.openrewrite.quark.Quark cannot be cast to class org.openrewrite.java.tree.J (org.openrewrite.quark.Quark and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`
- `class org.openrewrite. c a .quaS v Piaop)rtrk.Quark cannot be cast to class org.openrewrite.java.tree.J (org.openrewrite.quark.Quark and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`
- `class org.openrewrite.quark.Quark.r).nwrwtriv.ckg cannot be cast to class org.openrueewrite.java.tree.J (org.openrewrite.quark.Quark and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`
- `class org.openrewrite.xml.tree.Xml$Document cannot be cast to class org.openrewrite.java.tree.J (org.openrewrite.xml.tree.Xmw l toreueru.aaol$Document and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`
- `class org.openrewrite.properties.tree.Properties$File cannot be cast to class org.openrewrite.java.tree.J (org.openrewrite.properties.tree.Properties$File and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`
- `class org.openrewrite.text.PlainText cannot be cast to class org.openrewrite.java.tree.J (org.openrewrite.text.PlainText and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexc`
- `class oaavceaauaekna oa crg.openrewrite.quark.Quark cannot be cast to class org.openrewrite.java.tree.J (org.openrewrite.quark.Quark and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`
- `class org.openreneex Cwrite.quark.Quark cannot be cast to class org.openrewrite.java.tree.J (org.openrewrite.quark.Quark and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`
- `class org.openrewrin ute.quark.Quark cannot be cast ar ap JFT$aSjc to class org.openrewrite.java.tree.J (org.openrewrite.quark.Quark and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`
- `class oaFu  org.openrewrite.quark.Quark cannot be cast to class org.openrewrite.java.tree.J (org.openrewrite.quark.Quark and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`
- `class org.openrewrite.xml.tret 3tcrtr(nlce.Xml$Docume(aol.ant cannot be cast to class org.openrewrite.java.tree.J (org.openrewrite.xml.tree.Xml$Document and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`
- `class org.openrewtrite.xml.tree.Xml$Document cannot be cast to class org.openrenxJP c oorsa)t  Dwrite.java.tree.J (org.openrewrite.xml.tree.Xml$Document and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`
- `clael2uvuaas sa )e  ss org.openrewrite.json.tree.Json$Document cannot be cast to class org.openrewrite.java.tree.J (org.openrewrite.json.tree.Json$Document and org.openrewrite.java.tree.J are cin unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`
- `class org.openrewr(.lrite.xml.tree.Xml$Document canxnot be cast to class org.openrewrite.java.tree.J (org.openrewrite.xml.tree.Xml$Document and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm @5600a278)`

**Comment by timtebeek** [comment]
Hi @AlexanderSkrock ! Could you provide us with some more details how you're running OpenRewrite in this case? Is it a Maven or Gradle project for instance, and did you add anything to the build files or are you running without modifying the build?

There's something weird going on with the output above; a lot of the messages are malformed it seems; did you run anything in parallel perhaps that butchered the output?

**Comment by AlexanderSkrock** [comment]
Hi,

I am using Maven and there was no parallel task. But I agree with you, the output is scrambled up.. :/

Here's the stacktrace as requested:
`[DEBUG] 
jters.java:60eduleVisit$3[m java.utinamed dadava.lang.ClassCastException: quark.Quark cannot be cast to ehaus.plexus.classworlds.realm.ClassReal$mapAsync$0modulTask.java:137  l.concurrent.FoClassCastExceptionlass org.openrewrite.quark.Qua (R.openrewrite.java.cleanup.RemoveEmptyJavaDocParkJoinPool.java:1622)
    oveEmptyJavaDocParameters.java:55)
ss )
l.java:1655)
       ForkJoat org.openrewrite.java.cleanup.Rat java.util.concurrent.ForkJoinPool.runWorker    aorg.openrewrite.quark.    at org.openrewrite.RecipeScheduler.lambda$scheduleVisit$3 (Recat javameters$1.visitSourecipeScheduler.java:254)rk rk.Quark cannot be cast to class org.openrewrite.java: rkJoinPool.runWorkerat java.util.concurrent.For5)
  e of loader (Rm @411class org.openrewriclate.java.tree.J (o67ded)ecipeS org.codehaus.plexus.classworl  at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.jkJoinTask$AdaptedRunnableAct (ForkJoinPool.java:1622)
    atclass org.openrewrite.quark.Quark cannot be cast to class org.openrew.tree.J (org.and org.openrewrite
    ceFilea.uipeSchedulerQuark cannot be cast to class org.openrewrite.java.tree.J (org.openrewrit org.openrewrite.java.cleanup.RemoveEmptyJavaDocParameters$1 (FemoveEmptyJainPool.java:1622)
    at java.util.concurrent.ForvaDocParorkJoinPool.java:1622).visitSourte.quark.Quar.jatil.co (RemoveEmptyJavaDoat org.openre.java.tree.J aropenrewrite.java.tree.J (org.openrewrite.quark.Quark and org.op java.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorkerThread.javaion.execava:373)
    at java.utds.realm.ClassRealm @41167ded)che
rg.openrewrite.quark.Quark and org.openrewriss te.java.tree.J are in unnamed module of     at orduler.java:58)
    at or
    at org.openrewritil.concurrent.ForkJoinPool$WorkQueue.topLevelExec (ForkJoinPool.java:1182)
enrewrrite.quark.Quark and org.openre in unnamed module write.RecipeScheduler.lambda$mapAscParameters.java:60)ncurrent.ForkJoinWorkerThread.runva:254)
k and org.openrewrite.java.tree.J areceFile
    ameters$1.visitSourceFile (RemovekJoinWorkerThread.runEmptyJavaDocat java.util.concurrent.ForkJoinWorkerThread.run (RemoveEmptyJavaDocPa in unnamed module of     at org.openrewrite.RecipeSched (ForkJoinWorkerThread.java:
  ync$0of loader org.codehaus.plexus.cewrite.java.tree.J are in unnamed module of loader org.coite.java.tree.J are in unname
)
    at java.utilutil.concurrent.Fore.java.cleanup.RemoveEmptyJavg.openrewrite.scheduling.ForkJoinScheduler.lambda$schedule$0g.openrewrite.java.cleanup.Remoloader org.codorg.openrewrite.quark.Quark cannot be cast to class org.openrewrite.java.tree.J (org.openrewrite.quark.Quark and org.openrewrite.java.tree.J are in unnamed module of loader org.codehaus.plexus.clehaveEmptyJ (ForkJoinScheduleaDocParameters$1.visikJo.concurrent.ForkJoinTask.doExed module of loader org.codehaus.plexus.classworlds.realm.ClassRealm lasswo (Recip  at org.openrewrite.java.cleanup.Remov165)
uler.lambda$mapAsync$0loader org.crameters.java:55) (ForkJoinPar (ForkJoinWorkerThread.java:165ameters.java:55)
  Wor
    oat org.openrewrite.RecipeScheduler (ReeEmptyJavaDocParametereScheduler.java:58)
rlds.realm.ClassRealm @41167ded)
@41167ded)
dehaus.plexus.classc (ForkJoinTask.jainPool.scatSourceFilr.java:41)
    avaDocParameters$1.visitSourceFileus.plexuassworlds.realm.ClassRealm @411s.classworlds.realm.ClassRealm @41167ded) (RemoveEmptyat java.util.concurrent.Forke (RemoveEmptyJavaDocParameters.javn (ForkJva:373)
    at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelEworlds.rea    at org.openrewrite.java.cleanup.RemoveEmptyJavaDocPara    at org.o    ats$1.visitSourceFilecipeS.lambda$scheduleVisit$3dehaus.plkerThread.java:165)
  at org.openrewri)
te.RecipeSchedexus.class (Recheduler.j (Re org.openrewrite.scheduling.ForkJoinSchepenmeters$1.visitSourceFilelm.ClassRealm @41167ded)xecoinPoola:60)
    JoinTask$AdaptedRunnableAction.execJavaDocParameters.j
67ded)
    at org.openrewrite.java.cl    at org.openrewrite.jaava:60)
 (Fat org.java:1655)
     (ForkJoinPool.java:1182)
    at java.util.c
    at org.openrewrite.java.cleanup.RemoveEmptyJavaDocParameters$1.visitSourceFile (RemoveEmpty (RemoveEmptyJavaDocParameterewrite.java.cleanduler.lambda$schedule$0moveEmpava:58cipeScworlds.realm.ClassRealm @41167deuler.lambda$scheduleVisitd)
    heduler)
tyJ (ForkJoinScheduler.java:41)
up.RemoveEmptyJavaDocPars.java:60)
    at org.openrewrite.javJavaDocParameters.java:60oncat jav.openrorkJoinTask.java    at org.openrewrite.java.cleanup.Remova.cleanup.RemoveEmptyJavaDocParameters$1eanup..visiveEmptyJavaDocParameters$1.visitSourceFile:1375)
ewrite.java.cleanup.a.util.concurrent.ForkJoinPool.runWorker (ForkJoinurrent)
    at org.a.cleanup.RemoveEmptyJavaDocParameters$1.visitSourceFilerameters$1.visitSou    avaDocParameters.jav    .javat or$3g.openrewa:254)at org.a:55)
    atat java.util.concurrent.rceFile (RemoveEmptyJavaDocParameters.java:60)
    at org.openrewrite.java.cleanup.RemoveEmptyJavaDocParameters$1.visitSourceFile (RemoveEmptyJavaDocParameters.java:55)
    at org.openrewrite.RecipeScheduler.lambda$scheduleVisit$3 (RecipeScheduler.java:254)
    at org.openrewrite.RecipeScheduler.lambda$mapAsync$0 (RecipeScheduler.java:58)
    at org.openrewrite.scheduling.ForkJoinScheduler.lambda$schedule$0 (ForkJoinScheduler.java:41)
    at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec (ForkJoinTask.java:1375)
    at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:373)
    at java.util.concurrent.ForkJoinPool$WorkQueue.openrewrite.java.cleanup.RemoveEmptyJavaDocParameters$1.visitSourceFile (RemoveEmptyJavaDocParameters.java:55)
    at org.openrewrite.RecipeScheduler.lambda$scheduleVisit$3 (RecipeSchedu.ForkJoinPool.java:RemoveEmptyJavaDocParameters$1.visitSourceFile    at java.uti (RemoveEmptyJavaDocParameters.java:55tSourceFileRemoveEmptyJavaDocParameter (RemoveEmp)
    atl.concurrent (RemoveEmptyJavaDocParameters.java:55)
    at org.openrewrite.RecipeScheduler.lambda$scheduleVisit$31622)
    Pooler.java:2topLevelExec (ForkJoinTask$AdaptedRunnableAction.exec (ForkJoinTa org.opopenrewrite.sche
    rite.java.cleanu (RecipeScheduler.java:254)
    at org.openrewritep.Removat duling.ForkJoinScheduler.lambda$schedule$0 (ForkJoinScheduler.java:41)
    at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec (ForkJoinTask.java:1375)
    at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:373)
    enrewrite.RecipeSchedulesk.java:1375)RemoveEmptyJavaDocParameters.java:55 (ForkJ54)
    at org.openrewrite.Recl.scanat (Recipe.ForkJoinTask.doExec (ForkJoinTask.java:373)
    at java.util. ortyJavaDocParas$1.visitSourceFilemeterg.openrewrite.RecipeScheduler.lambda$scheduleVisit$3concurrent.ForkJoinPool$WorkQueue.topLevelEScheduler.java:254)
    at org.openrewrite.RecipeSc java.util.concurrent.ForkJoinWorkerThread.run  (ForkJoinWorkipeScheduoinPoo)
   r.lambda$scheduleVisit$3atorg.openrewrite.RecipeScheduler.lambda$mapAsync$0 (RecipeScheduler.java:58)
 at java.util.concurrent.ForkJoinTask.doExec (Fo    at org.openrewrite.RecipeScheduler.lambda$schel.java:118ler.lambda$mapAsync$0erThread.java:165)
(ForkJoinPool.java:16heduler.lambda$mapAsync$0xec (s.java:60)
    at org.openrewrite. (RemoveEmptyJavaDocParameters.java:60)
    java.cleaRecipeSc (ForkJoinPool.java:1182)
     (RecipeScheduler.java:58)
    at org.openrewrite.scheduling55) (RecipeScheduler.java:58)
    2)
    at java.util.concduleVisit$3rkJoinTask.java:373)
    at java.util.c
    at org.openrewrite.RecipeScheduler.lambda$mapAsync$0 (RecipeSchedurkJinScheduler.lambda$schedule$0:60 (RecipeScheduler.java:58)
    )
 (ForkJoinScheduler.java:41)
    aoinPool$WorkQler.java:58)oncu (RecipeScheduler.java:254)urrent.ForkJoinPool.scanat org.openrewrite.scheduling.ForkJoinSchedu
    .ForkJoinScheduler.lambda$schedule$0at java.util.concurrent.ForkJoinPool.scanheduler.java:254)
    at org.openrewrinup.Remat org.openreoveEmptyJavaDote.RecipeScheduler.lambda$mapAsync$0 ( (ForkJoinPoo (Forkatler.lambda$schedule$0 (ForkJoinScheduler.java:41)
    at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec (ForkJoinTask.java:1375)
     (ForkJoinPool.java:1655)
    rrent.ForkJoinPool$WorkQueue.t
    ueue.topLevelExec (Fot ja    at org.openrewrite.java.cleanup.RemoveEmatptyJavaDocParameters$1.visitSourceFileva.util.concurrent.ForkJoinTask$AdaptedRunnrkJatopLat org.openrewrite.RecipeScheduler.lambda$mapAsync$0 (RecipeSche
    at java.util.concurrent.ForkJoinPool.runWorkerJoinScheduler.java:41)
    at jl.java:1655)
    RecipeScheduler.java:58)
    at ocParameterswri$1.visitSourceFile (rg.openrewrite.scheduling.ForkJoinScheduler.lambda$schedule$0at java.util.concurrent.ForkJoinPool.runWorkerava.util.concurrent.ForkJoinTask$AdaptedRunnableAct (ForkJoinPool java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:373)
at java.util.concurrent.ForkJoinPool.runWorkerdulerevelExec org.openrewrite.schoinPoolableAction.exec (Remo org.openrewrite.scveEmptyJavaDocParameters.java:55 (F.java:1182)
    at java.util.concurrent.ForkJoieduling (ForkJoinPool.java:1182).java:58)
    a (ForkJ    at java..java:1622ion. (ForkJoinPool.java:1 (ForkJoinScheduleRemoveEmptyJavaDocParameters.java:55)
te.java.cleanup.RemoveEmptyJavaDocParameters$1.visit    r.j622)
    atexec ()
util.concuroinPool.t org.openrewrite.scheduling.ForkJoinScheduler.lambda$schedule$0 (ForkJoi
  .ForkJnPool.scano (ForkJoinPool.jav)
    heduliang.ForkJoinScheduler.lambda$scheda:1655)
  rkJoinTask.java:1375)
rent.ForkJoinPool$WorkQueue.topLevelExec    at java.util.concurrent.ForkForkJoinTask.java:137 java.util.concurrent.ForkJoinWorkerThread.runava:41)
    at java.util.concurrent.ForkJoinTask$Aat org.openrewrite.RSourceFileecipedapt (F5)
    JoinWorkerThread.run (F
    at java.ut41at java.util.concurrent.Fer.lambda$schedule$0at java.util.concurrent.ForkJoinTask.doExec  at java.util.concurreule$0t org.openrewrite.RecipeScheduler. (ForkJnt.ForkJoinPool.runWorker (ForkJoinTask.java:373)
   (ForkJoinScheduleorkJo)
    il.concurrorkJoinPool.java:1182) (ForkJoinWorkerThread.java:165)
at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:373)
    aorkJoinWorkerThread.java:165)
edRunnableActioScheduler.lambda$scheduleVisi (Ret$3 (RecipeScheduler.java:254)
    at org.openrewrite.RecipeScheduler.lambda$mapAsyncn.exect java.u
  ent.ForkJoinWorkeratinPool.scanr.java  at (FoinScheduler.java:41)lambda$scheduleVisit$3
 java.utThread.run (ForkJoinWorkerThre  at java.util.concurtil.concurrent.Fork (ForkJoinTask.java:1375)
    $0moveE (RecipeScheduler.java:58)
    at java.util.concJoinPool$WorkQueue.topLevelExerent.ForkJoinPool.scanad.java:165)il.concurrent.ForkJoinTask$AdaptedRunnableAction.exec
    at java.util.concurrent.Fo)
    al$WoJoinPool.java:1622)
dRunnableAction.exec (RecipeScheduler.java: (ForkJoinTask.j    at java.util.concurrent.ForkJoinWorkQut java.util.concurrent.ForkJoinPool.runWorker (ForkJoinTask.java:1375)
 (ForkJoinPool.java:1655)
    c (ForkJoinPool.java:1182)
    at java.util.concurrent.ForkJoinPool.scan (ForkJoinPoolurrent.ForkJoinTask.doExecat org.openrewrite.scmptyJavaDocParameters.java:55)
    at org.openrewrite.RecipeScheduleheduling.ForkJoi (ForkJoinTask..java:1655)
    at java.util.concu    at java.util.concurrent (ForkJoinPool.java:1622)
    at jrkJoinTask$AdaptedRunnableAction.exec (ForkJoinTask.java:1375)
    at java.util.concueue.topLevelExecrkerThread.runava:1375)
    at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:373)
    at254)
    at org.openrewrite.Rec java.util.concurrent.ForkJ (ForkJoinWorkerThread.jav (ForkJoinPoolrrent.ForkJoinTask.doExecava.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorkerThread.java:165)
.ForkJoinTask.doExerrent.Forkat java.util.concurrejava:373)
    nScheduler.lambda$scher.lambda$scheduleVisit$3 (RecipeScheduler.java:254)
    dule$0at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelEnt.ForkJoJoinPool.runWorkerc (ForkJoinPool.java:1622)
    at java.util.concurrent.ForkJoinWorkerThread.r (ForkJoinTask.java:373)
    at java.util.concurrent..java:1182)
a:165)
oinPipeScheduler.lool$WorkQueue.topLevelExec (ForkJoinPool.java:1182)
      at java.util.concurrent.ForkJoinPool.sForkJoinPool$WorkQueue.topLeveun (ForkJinPool.runWorkerxec (ForkJoinPool.java:1182)
    at java.util.concurrent.ForkJoinPool.scan (ForkJoinPool.java:1655)
    at java.util.concurrent.Fo (ForkJoinScheduler.jaat org.openrewrite.RecipeScheduler.lambda$mapAsync$0 (RecipeScheduler.java:58)
 va:41)
    rkJoinPool.runWorker (Fo (ForkJoinPool.java:1622)
    aoinTask.java (ForkJoinWorkerlExec (ForkJoinPool.java:1182can  at java.util.concurrent.ForambkJoinPool.scan (ForkJoi)
Threa:37t java.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorkJoinPool.java:1622)
    atat java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec   at org.openrewrite.schedu (ForkJoinTask.java:1375)
    at java.util.concurrent.ForkJoinTask.doExec jarkerThread.3)
    d.java:165   nPool.java:1655)
 (ForkJoinPool.jda$mapAsyncava:16     at java.util.concurre)
at java.util.cjava:165)
va.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorke (Forkling.FJoinTask.java:373)
    rThread.java:165)
oncurrent.ForkJoinPool$WorkQueue.topnt.at java55)$0
    .util.concurrent.ForkJoinPool.ForkJoinPool.scan (ForkJoiLevelExecat java.uorkJoinScheduler.lambda$schedule$0til.concurrent.ForkJoinP (ForkJnPool.java:runWorkerat java.util.concurrent. (ReciForkJo (Fork1655)
oinPool.java:11ool$WorkQueue.topLevelExec (ForkJoinSc (ForkJoinPool.java:1182)
    82)
    at    at java.util.JoinPooinPool.runWpeScheduler.orkerl.java:1concurrent.ForkJoinPool.runWor java.util.concurrent.ForkJoinPool.scanat java.util.concurrent.ForkJoinPool.scan (ForkJoinPool.java:1655)
    at java.util.concurrent.ForkJoinPool.runWorker (ForkJoheduler.java:41)
 inPool.ja (Fker622)
java:58)
    at java.util.concur.util.concurrent.ForkJoinWorkerThread.run (ForkJoinPool.jaorkJoinPool.va:162   2)
2)
    at java.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorkerThread.java)
  at java.util.concurrent.ForkJoinWorkerThread.runat java.ut (ForkJoinWorkerThread.    a:165)
 (ForkJoinWorkerThread.java:165)
nrewrite.scheduling.ForkJoinScheduler.lat java.java:165)
il.concurrent.ForkJoinTautil.concurrent.ForkJoinPool.rumbda$schedule$0nWosk$AdaptedRunnableActiorker (ForkJoinPool.java:1 (ForkJoinScheduler.java:41)
    at java.util.concurrent.Fo622)
n.exec    arkJt java.util.concurrent.ForkJoinWorkerT (Forhread.runoinTask$AdaptedRunna (ForkJoinWorkerThread.java:165)
kJoinTask.java:1375)
  bleAction.exec  at java.util.concurrent.ForkJo (ForinTask.doExec (ForkJoinTask.java:373)
    at kJoinTask.java:1375)
  java.ut  at java.util.concurrent.il.concurrent.ForkJoinPool$WorkQueuForkJoinTask.doExece.top (ForkJoinTask.java:373)
 LevelExec    at java.util.concu(ForkJoinPool.java:1182)
    at java.util.concurrent.ForkJorrent.FinPool.scanorkJo (ForkJoinPool.jainPool$WorkQueue.topLevelExec (ForkJoinPool.java:1655)
    at java.util.concurrent.ForkJoinPoova:1182l.runWorker (ForkJoinPool.)
    java:at java.uti1622)
    at java.util.col.concurrent.ForkJoinPooncurrent.ForkJoinWorkerThread.runl.scan (ForkJoin (ForWorkerThread.java:165)
kJoinPool.java:1655)
    at java.util.concurrent.ForkJoinPool.runWorker (ForkJoinPool.java:1622)
    at java.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorkerThread.java:165)
[ERROR] The recipe produced an error. Please report this to the recipe author.`

**Comment by AlexanderSkrock** [comment]
I run OpenRewrite using the following command: `rewrite:run -e -X -D skip.webpack -D skip.yarn -D skip.karma -f pom.xml`.

Sure, I can provide at least share the build configuration:
```xml
       <plugins>
            <plugin>
                <groupId>org.bsc.maven</groupId>
                <artifactId>maven-processor-plugin</artifactId>
                <version>3.3.3</version>
                <executions>
                    <execution>
                        <id>process</id>
                        <goals>
                            <goal>process</goal>
                        </goals>
                        <phase>generate-sources</phase>
                        <configuration>
                            <outputDirectory>target/metamodel</outputDirectory>
                        </configuration>
                    </execution>
                </executions>
                <dependencies>
                    <dependency>
                        <groupId>org.hibernate</groupId>
                        <artifactId>hibernate-jpamodelgen</artifactId>
                        <version>${hibernate.version}</version>
                    </dependency>
                </dependencies>
            </plugin>
            <plugin>
                <groupId>org.codehaus.mojo</groupId>
                <artifactId>build-helper-maven-plugin</artifactId>
                <executions>
                    <execution>
                        <id>add-source</id>
                        <phase>generate-sources</phase>
                        <goals>
                            <goal>add-source</goal>
                        </goals>
                        <configuration>
                            <sources>
                                <source>target/metamodel</source>
                            </sources>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.openrewrite.maven</groupId>
                <artifactId>rewrite-maven-plugin</artifactId>
                <version>4.43.0-SNAPSHOT</version>
                <configuration>
                    <!--<skipMavenParsing>true</skipMavenParsing>-->
                    <configLocation>${maven.multiModuleProjectDirectory}/rewrite.yml</configLocation>
                    <activeRecipes>
                        <recipe>org.openrewrite.java.JavaRecipes</recipe>
                        <!--<recipe>org.openrewrite.java.testing.junit5.JUnit5BestPractices</recipe>-->
                    </activeRecipes>
                </configuration>
                <dependencies>
                    <dependency>
                        <groupId>org.openrewrite.recipe</groupId>
                        <artifactId>rewrite-migrate-java</artifactId>
                        <version>1.17.2</version>
                    </dependency>
                    <!--<dependency>
                        <groupId>org.openrewrite.recipe</groupId>
                        <artifactId>rewrite-testing-frameworks</artifactId>
                        <version>1.35.0-SNAPSHOT</version>
                    </dependency>-->
                </dependencies>
            </plugin>
        </plugins>
    </build>
```
In addition to that, we use the `maven-tiles-plugin` with some tiles of our company:
```xml
<plugin>
    <groupId>io.repaint.maven</groupId>
    <artifactId>tiles-maven-plugin</artifactId>
    <version>2.34</version>
</plugin>
```
Also, I had to exclude a dependency "artemis-jms-client" because they configured an invalid windows patch as relative path and while pom parsing it would otherwise fail. See: https://github.com/apache/activemq-artemis/blob/main/pom.xml#L29, but that is another issue, which is rather on their end in my opinion.

Kind regards
Alexander Skrock

**Comment by AlexanderSkrock** [comment]
Oh, I just commented out `org.openrewrite.java.cleanup.RemoveEmptyJavaDocParameters` from my yaml configuration file and it suddenly worked. That stacktrace gave me the hint. So it is definitely about that recipe.

**Comment by knutwannheden** [comment]
@AlexanderSkrock Thank you for your analysis! We will look into a fix for this.

**Comment by AlexanderSkrock** [comment]
Thank you both, @timtebeek  and @knutwannheden for the quick responses and resolution!
